packages/block-library/src/navigation-submenu/edit.js

```diff
@@ -350,8 +348,8 @@ export default function NavigationSubmenuEdit( {
 
 			return {
 				isAtMaxNesting:
-					getBlockParentsByBlockName( clientId, name ).length >=
-					MAX_NESTING,
+					getBlockParentsByBlockName( selectedBlockId, name )
```

Comment:
Would you elaborate on changing `clientId` to `selectedBlockId` here? Would it make sense to apply the same change in line 353, 356, 361 below?

Reply:
For calculating nesting and maxNesting we need to start getting parents from the currently selected navigation link block, not from the submenu itself. I think the other flags below apply to the submenu block not to the navigation link items.

Reply:
Hm it looks like the `pull()` like is triggered for any selected block, even if it isn't a child block of the currently selected navigation submenu. When I added two navigation blocks, I could not add any submenus inside the one **without** max nesting set:

https://user-images.githubusercontent.com/205419/161282312-c0f635e2-8f18-4299-9945-4ae5062d03ef.mp4

Reply:
ah snap

Reply:
But great catch, for some reason I didn't think to test with a second navigation 🤦🏻

Reply:
That's a great catch. I wonder if we have any more bugs like that with x2 Nav blocks at once?

Reply:
Perhaps 🤷  This one just stood out to me because of how `pull()` was called at the top level – I thought "hmm but that would also be called in a second instance, no?"

Reply:
@adamziel you basically solved [this ](https://github.com/WordPress/gutenberg/pull/38621#issuecomment-1065027235) the right way. By switching to selectedBlockId I ignored the fact that the allowed blocks global was mutated for every instance. Excellent debugging.

Not mutating that global solves the problem I "fixed" with checking the `selectedBlockId`. 👏🏻